src/chaincode/dex/addLiquidity.ts

```diff
@@ -52,6 +53,10 @@ export async function addLiquidity(
 
   const key = ctx.stub.createCompositeKey(Pool.INDEX_KEY, [token0, token1, dto.fee.toString()]);
   const pool = await getObjectByKey(ctx, Pool, key);
+
+  // Check private pool access
+  validatePrivatePoolAccess(pool, ctx.callingUser);
```

Comment:
since the method throws an exception if the validation fails, maybe it could have a stronger name, like `ensureUserCannAccess()`, or `validateOrRejectPoolAccess`, or similar